src/v0/destinations/amazon_audience/networkHandler.js

```diff
@@ -0,0 +1,139 @@
+const {
+  NetworkError,
+  ThrottledError,
+  AbortedError,
+  RetryableError,
+} = require('@rudderstack/integrations-lib');
+const { prepareProxyRequest, handleHttpRequest } = require('../../../adapters/network');
+const { isHttpStatusSuccess } = require('../../util/index');
+const {
+  processAxiosResponse,
+  getDynamicErrorType,
+} = require('../../../adapters/utils/networkUtils');
+const { REFRESH_TOKEN } = require('../../../adapters/networkhandler/authConstants');
+const { DESTINATION, CREATE_USERS_URL, ASSOCIATE_USERS_URL } = require('./config');
+const { TAG_NAMES } = require('../../util/tags');
+
+const amazonAudienceRespHandler = (destResponse, stageMsg) => {
+  const { status, response } = destResponse;
+
+  // to handle the case when authorization-token is invalid
+  // docs for error codes: https://advertising.amazon.com/API/docs/en-us/reference/concepts/errors#tag/Audiences/operation/dspCreateAudiencesPost
+  if (status === 401) {
```

Comment:
Rather than only checking with status code can we also match with Message to know when exactly should we refresh. As I see there are other scnearios for 401

Reply:
I checked for it but message body along with response is same for all 401's so there is nothing else we can do.
Keeping a note `in case of 410's check for cases of 401's` in docs